util/tokens.py

```diff
@@ -0,0 +1,12 @@
+SOURCE_AUTHORIZATION_HEADER = 'X-Authorization-Source'
+SOURCE_ENV_VARIABLE = 'SOURCE_SLACK_API_TOKEN'
+DESTINATION_AUTHORIZATION_HEADER = 'X-Authorization-Destination'
+DESTINATION_ENV_VARIABLE = 'DESTINATION_SLACK_API_TOKEN'
+
+def token_from_header_or_env(token):
```

Comment:
weird name considering it knows nothing about headers or env or where tokens are from

Reply:
maybe "format token"?

where does the `Token: ` string come from?

Reply:
From the client 😉.  It's following HTTP [syntax conventions](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization#Syntax) for indicating the type of an auth header.  Trying to communicate that "this is an auth header but I'm using `X-Auth-*` because it's not actually authorization for this client".

Looking at slack's thing, maybe I should use "Bearer":

https://github.com/slackapi/python-slackclient/blob/master/slack/web/base_client.py#L67

Reply:
It knows that the header will begin with `Token ` is basically all it does.  I didn't want to re-write a bunch of ifs.  I was previously one-lining at:

```py
request.headers.get(HEADER_NAME, os.environ.get(ENV_VAR_NAME, '')).replace('Token ', '')
```

But felt that it was a bit obtuse.